### PR TITLE
Refactored cephcluster_tests.go

### PR DESCRIPTION
**Pull request by priyanka19-98**
Instead of explicitly writing each test case within a function, it is reorganized to follow a table driven approach. The refactoring is done for `cephClusterCreate, cephClusterUpdate,cephClusterWithNoConditions, cephClusterWithNegativeConditions`

Signed off by: Priyanka Jiandani <pjiandan@redhat.com>

### CODE REVIEW

**Comment by priyanka19-98** [comment]
@jarrpa Please review

File: pkg/controller/storagecluster/cephcluster_test.go [hunk 1]
```diff
@@ -13,68 +13,84 @@ import (
...
-
-	expected := newCephCluster(mockStorageCluster, "", 3, log)
-	actual := newCephCluster(mockStorageCluster, "", 3, log)
-	err = reconciler.client.Get(nil, mockCephClusterNamespacedName, actual)
-	assert.NoError(t, err)
-	assert.Equal(t, expected.ObjectMeta.Name, actual.ObjectMeta.Name)
-	assert.Equal(t, expected.ObjectMeta.Namespace, actual.ObjectMeta.Namespace)
-	assert.Equal(t, expected.Spec, actual.Spec)
-}
+func TestEnsureCephClusterSystem(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (priyanka19-98) [reply pkg/controller/storagecluster/cephcluster_test.go]: Done!

File: pkg/controller/storagecluster/cephcluster_test.go [hunk 2]
```diff
@@ -91,59 +91,73 @@ func TestEnsureCephClusterSystem(t *testing.T) {
 	}
 }
 
-func TestStorageClusterCephClusterCreation(t *testing.T) {
-	// if both monPVCTemplate and monDataDirHostPath is provided via storageCluster
+func TestStorageClusterCephCluster(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (priyanka19-98) [reply pkg/controller/storagecluster/cephcluster_test.go]: Done!

File: pkg/controller/storagecluster/cephcluster_test.go [hunk 3]
```diff
@@ -91,59 +91,73 @@ func TestEnsureCephClusterSystem(t *testing.T) {
...
-	assert.Equal(t, generateNameForCephCluster(sc), actual.Name)
-	assert.Equal(t, sc.Namespace, actual.Namespace)
-	pvcSpec := actual.Spec.Mon.VolumeClaimTemplate.Spec
-	assert.Equal(t, mockDeviceSets[0].DataPVCTemplate.Spec.StorageClassName, pvcSpec.StorageClassName)
-	assert.Equal(t, "/var/lib/rook", actual.Spec.DataDirHostPath)
+	cases := []struct {
+		label                    string
+		sc                       *api.StorageCluster
+		monPVCTemplate           *corev1.PersistentVolumeClaim
+		monDataDirHostPath       string
```
[the remark this answers is not in the document]
Comment (priyanka19-98) [reply pkg/controller/storagecluster/cephcluster_test.go]: Done!

File: pkg/controller/storagecluster/cephcluster_test.go [hunk 4]
```diff
@@ -91,59 +91,73 @@ func TestEnsureCephClusterSystem(t *testing.T) {
...
+		expctdMonDataDirHostPath string
```
[the remark this answers is not in the document]
Comment (priyanka19-98) [reply pkg/controller/storagecluster/cephcluster_test.go]: Done!

File: pkg/controller/storagecluster/cephcluster_test.go [hunk 5]
```diff
@@ -13,70 +13,90 @@ import (
...
-	assert.NoError(t, err)
-	assert.Empty(t, reconciler.conditions)
-}
-
+<<<<<<< HEAD
 func TestStorageClusterCephClusterCreation(t *testing.T) {
 	// if both monPVCTemplate and monDataDirHostPath is provided via storageCluster
+=======
+func TestNewCephClusterMonData(t *testing.T) {
+>>>>>>> 81b22226... Changed function names
```
Comment (jarrpa) [inline pkg/controller/storagecluster/cephcluster_test.go]: You missed a conflict here. You want to accept the old version (`TestStorageClusterCephClusterCreation()`) and then apply the new version in a later commit. You can accomplish this during a `git rebase -i` by editing the commit to remove everything except line 95, then resolving any conflicts in later commits.

Comment (priyanka19-98) [reply pkg/controller/storagecluster/cephcluster_test.go]: Fixed the conflict. Let me know if any further changes are required. Thanks

**Pull request merged**